### ISSUE TITLE
How do I use the pre-trained model to run on my own Images.

### ISSUE THREAD

**Issue by Santakid** [seva100/optic-nerve-cnn#13]
I can run your U-Net, OD on DRIONS-DB (fold 0).ipynb with no problem using your pre-trained model. And for the dataset I didn't recreate it, I downloaded from the url you gave. So now I want to run this model on other images but i don't know how. Do I have to start at the beginning of recreating dataset just to run 1 of my image? Please help, I am new to coding

**Comment by seva100** [comment]
@Santakid, no you don't need to recreate the dataset. Actually you can just load a network with pretrained weights, load your picture, do a CLAHE and feed to the network.

**Comment by seva100** [comment]
https://github.com/seva100/optic-nerve-cnn/issues/10#issuecomment-568898553 -- here I provide some inference code in the comment which you can use.

**Comment by Santakid** [comment]
Thank you so much and sorry for not realizing that there is a similar question that has already been answered. But I still have one more problem is that when I load in the exact same image you did in the network and when I compare it to its respective corrected image, I don't get the same percentage difference as you did. I think this is because I am comparing it with the corrected image that has gone through the dataset process. 
`cur_iou = K.eval(mean_IOU_gpu(prediction[None, None, ...], corr[None, None, ...]))`
I don't know which variables is the corrected mask of the original image to replace corr in above code.

**Comment by seva100** [comment]
Hm, I'm not sure I fully understood your question. If you run the network as it is, do you get the same IOU score for an image?
If a score is only different by a very small value (< 1.0), this can be attributed to some changes in libraries versions.

**Comment by Santakid** [comment]
I load in the Original image from data/DRIONS-DB/images and it got a IOU score of 0.6378245 but yours got 0.92216 , This is the code I ran.
`img_path` = '/content/drive/My Drive/optic-nerve-cnn/data/DRIONS-DB/images/image_001.jpg'
im = np.array(Image.open(img_path))
im = im[0:, 40:]
print(im.shape)

#Image
im = cv2.resize(im, (256,256))
im = im.astype(np.float64) / 255.0
im = skimage.exposure.equalize_adapthist(im)
print(im.shape)
plt.imshow(im), plt.show()

#Predicted Image
im = np.expand_dims(im, axis=0)
im = tf_to_th_encoding(im)
OwnPred = (model.predict(im)[0, 0]).astype(np.float64)
plt.imshow(OwnPred, cmap=plt.cm.Greys_r), plt.show()`

**Comment by Santakid** [comment]
I fixed it by comparing with original ground truth from optic-nerve-cnn/data/DRIONS-DB/DRIU results/DRIU/ instead.
